dbt-adapters/src/dbt/include/global_project/macros/materializations/tests/test.sql

```diff
@@ -1,6 +1,12 @@
 {%- materialization test, default -%}
 
   {% set relations = [] %}
+  {% set limit = config.get('limit') %}
+
+  {% set sql_with_limit %}
+    {{ sql }}
+    {{ "limit " ~ limit if limit != none }}
```

Comment:
won't this break on platforms that don't support limit?

Reply:
Can you use the `get_limit_sql` macro here?

Reply:
example where fabric overrides to achieve correct limit https://github.com/microsoft/dbt-fabric/blob/772d896d366137ff1ed1a364110d9eee67a727e6/dbt/include/fabric/macros/adapters/show.sql#L4

Reply:
Good call! 🧠 I'll make that change in this PR.

I'll save it for a separate PR to update `default__get_test_sql` to use [`get_limit_sql`](https://github.com/dbt-labs/dbt-adapters/blob/6255adf01aa4e3004d0267047c02813ee10d6b5e/dbt-adapters/src/dbt/include/global_project/macros/adapters/show.sql#L21):

https://github.com/dbt-labs/dbt-adapters/blob/6255adf01aa4e3004d0267047c02813ee10d6b5e/dbt-adapters/src/dbt/include/global_project/macros/materializations/tests/helpers.sql#L12

Reply:
Addressed in 421970712483177c49442bc9dba6f1c53dac4c51.

Felt awkward to use a macro named `get_limit_subquery_sql`, so opened https://github.com/dbt-labs/dbt-adapters/issues/1038 as a result.